pkg/handler/enqueue_mapped.go

```diff
@@ -79,10 +82,23 @@ func (e *enqueueRequestsFromMapFunc) Generic(ctx context.Context, evt event.Gene
 
 func (e *enqueueRequestsFromMapFunc) mapAndEnqueue(ctx context.Context, q workqueue.RateLimitingInterface, object client.Object, reqs map[reconcile.Request]empty) {
 	for _, req := range e.toRequests(ctx, object) {
-		_, ok := reqs[req]
-		if !ok {
-			q.Add(req)
-			reqs[req] = empty{}
+		if _, ok := reqs[req]; ok {
+			// reqs is a map of requests to avoid enqueueing the same request multiple times.
+			continue
+		}
+		// If the request doesn't specify a cluster, use the cluster from the context.
```

Comment:
nit: comment outdated (we don't take it from context anymore)

Reply:
fixed